src/dz17/second/Main.java

```diff
@@ -0,0 +1,20 @@
+package dz17.second;
+
+import java.util.List;
+
+import static dz17.second.Products.products;
+
+
+public class Main {
+    Product product;
+
+    public static void main(String[] args) {
+        List<Product> productList = products();
+        takeProduct(productList,10,"Book");
+    }
+    public static void takeProduct(List<Product> products, int sale, String type) {
+        products.stream().filter(fr -> fr.getType().equals(type))
+                .map(fr->fr.getType()+" preisec with sale= "+(fr.getPrices()-fr.getPrices()*sale/100))
```

Comment:
You transforming your product into `String`. In requirement asked to get list of products with discounts
`map(product->new Product(...,product.getPrice()*sale/100,...))`